src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift

```diff
@@ -20,29 +20,66 @@
  *  along with this program.  If not, see http://www.gnu.org/licenses/.
  */
 import ActiveLabel
+import UIKit
 
 @objc extension ProjectDetailStoreViewController {
 
     static var height: CGFloat = Util.screenHeight()
-    static var marginLeftPercentage: CGFloat = 15.0
+    static var inset: CGFloat = 25.0
+    static var buttonHeight: CGFloat = 25.0
+    static var spaceBetweenButtons: CGFloat = 5.0
+    static var labelHeight: CGFloat = 25.0
+    static var verticalNameLabelPadding: CGFloat = 5.0
+    static var verticalLabelPadding: CGFloat = 10.0
+    static var verticalSectionPadding: CGFloat = 20.0
+    static var loadingButtonSize: CGFloat = 28.0
+
+    static var fontSizeTitle: CGFloat = 17.0
+    static var fontSizeInformationItem: CGFloat = 14.0
+    static var fontSizeLabel: CGFloat = 12.0
+
+    func loadProject(_ project: CatrobatProject) {
+        self.projectView?.removeFromSuperview()
+        self.projectView = self.createProjectDetailView(project, target: self)
+
+        if self.project.author == nil {
+            self.showLoadingView()
+            let button = self.projectView.viewWithTag(Int(kDownloadButtonTag)) as! UIButton
+            button.isEnabled = false
+        }
+
+        self.scrollViewOutlet.addSubview(self.projectView)
+        self.scrollViewOutlet.delegate = self
+
+        if let reportButton = view.viewWithTag(Int(kReportButtonTag)) {
+            let height = reportButton.frame.origin.y + reportButton.frame.size.height + type(of: self).inset
+            self.scrollViewOutlet.contentSize = CGSize(width: self.scrollViewOutlet.frame.width, height: height)
+        }
+
+        self.scrollViewOutlet.contentInsetAdjustmentBehavior = .never
+        self.scrollViewOutlet.isUserInteractionEnabled = true
+    }
 
     func createProjectDetailView(_ project: CatrobatProject, target: Any?) -> UIView? {
-        let view = UIView(frame: CGRect(x: 0, y: 0, width: Util.screenWidth(), height: 0))
+        let view = UIView(frame: CGRect(x: 0, y: 0, width: Util.screenWidth(), height: CGFloat.greatestFiniteMagnitude))
         view.backgroundColor = UIColor.clear
-        view.autoresizingMask = .flexibleHeight
-        self.addNameLabel(withProjectName: project.projectName, to: view)
-        self.addAuthorLabel(withAuthor: project.author, to: view)
-        self.addThumbnailImage(withImageUrlString: project.screenshotSmall, to: view)
-        self.addDownloadButton(to: view, withTarget: target)
-        self.addLoadingButton(to: view, withTarget: target)
-        self.addOpenButton(to: view, withTarget: target)
+
+        let thumbnailView = self.addThumbnailImage(withImageUrlString: project.screenshotSmall, to: view)
```

Comment:
Maybe we should use screenshotBig here, the small one looks extremely blurry on larger devices

Reply:
Great idea, done!